python/lsst/daf/butler/datastore/file_templates.py

```diff
@@ -522,6 +531,24 @@ def format(self, ref: DatasetRef) -> str:
         output = ""
 
         for literal, field_name, format_spec, _ in parts:
+            if field_name and "|" in field_name:
+                alternates = field_name.split("|")
+                for alt in alternates:
+                    if "." in alt:
+                        primary, _ = alt.split(".")
```

Comment:
In theory we could extend the test here so that if some asked for `exposure.x|exposure` (say, because `x` was new in the universe but if it wasn't there fall back to `exposure`) fairly simply. I'm not sure if it's worth doing though.

Reply:
If we don't have a use case let's not do it.  Even though I buy that the approach you've taken on this ticket is the minimal change to solve the problem at hand, I'd still like to search in the future for ways to simplify all of this by offering less functionality, as opposed to extending that functionality further.